bot/bot.go

```diff
@@ -13,13 +13,54 @@ import (
 	"github.com/18F/angrytock/messages"
 	"github.com/18F/angrytock/slack"
 	"github.com/18F/angrytock/tock"
+	"github.com/nlopes/slack"
 )
 
+type readChannel struct {
+	key      string
+	response chan string
+}
+
+type writeChannel struct {
+	key      string
+	value    string
+	response chan bool
+}
+
+type mapStateManager struct {
+	state        map[string]string
+	readChannel  chan *readChannel
+	writeChannel chan *writeChannel
+}
+
+func newMapStateManager() *mapStateManager {
+	return &mapStateManager{
+		make(map[string]string),
+		make(chan *readChannel),
+		make(chan *writeChannel),
+	}
+}
+
+func (manager *mapStateManager) startMapStateManager() {
+	go func() {
+		for {
+			select {
+			case read := <-manager.readChannel:
+				read.response <- manager.state[read.key]
+			case write := <-manager.writeChannel:
+				log.Println(write)
+				manager.state[write.key] = write.value
+				write.response <- true
+			}
+		}
+	}()
+}
+
```

Comment:
if you want to dive deep into the rabbit hole of encapsulation, unexport UserEmailMap
then do 
```go
func (b *bot) WriteUserData (data *ReadChannel) {
   // use the write i just mentioned earlier. 
   b.userEmailMap.Write(data)
}
```